apps/microsoft-teams/app-actions/src/helpers/get-channels-list.ts

```diff
@@ -0,0 +1,54 @@
+import { AppActionCallResponse, Channel, TeamInstallation } from '../types';
+
+export const getChannelsList = async (
+  botServiceUrl: string,
+  apiKey: string,
+  tenantId: string
+): Promise<Channel[]> => {
+  const res = await fetch(`${botServiceUrl}/api/tenants/${tenantId}/team_installations`, {
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+      'x-api-key': apiKey,
+    },
+  });
+
+  // TODO: Parse the response instead of the assertion here
+  const response = (await res.json()) as AppActionCallResponse<TeamInstallation[]>;
+
+  if (!response.ok) {
+    throw new Error(response.errors?.[0]?.message ?? 'Failed to get channels');
+  }
+
+  const channelsList = transformInstallationsToChannelsList(response.data, tenantId);
+  return channelsList;
+};
+
+export const transformInstallationsToChannelsList = (
```

Comment:
I mentioned the idea of converting the `getChannelsList` function to a method of an API client. But I think if we do that it's better to keep that method "dumb" (just return the data without transforming it to our own data model in the app action).

but that means we _should_ keep this transformation function here as a helper and then take the list we get back and transform it to the format we need.